File: ROS/catkin_ws/src/my_robot_tutorials/src/scripts/robot_news_radio_transmitter.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String #import the message type to be used

if __name__ == '__main__':

	# initialize the node with node name as argument
	rospy.init_node('robot_news_radio_transmitter', anonymous=False)

	# create a publisher
	# arguments: name of topic, type off topic, buffer size
	pub = rospy.Publisher("/robot_news_radio", String, queue_size=10)

	# create a rate to make the loop wait for the specified time
	rate = rospy.Rate(2)	

	while not rospy.is_shutdown(): # while shutdown command is not given
		# create the instance of the message type object
		msg = String()
		# put in the data
		msg.data = "Hi, this is Radio Mirchi!"
		# publish the data
		pub.publish(msg)
		# pause the loop
		rate.sleep()

	# print a message at the end of node
	rospy.loginfo("Node has stopped")

File: ROS/catkin_ws/src/my_robot_tutorials/src/robot_news_radio_transmitter.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char **argv) {

	// create a node
	// arguments: argc, argv, name of node
	ros::init(argc, argv, "robot_news_radio_transmitter", ros::init_options::AnonymousName);
	// initialize node handles
	ros::NodeHandle nh;

	// create a ros publisher
	// make a nodehandle object advertise<message type>(topic name, buffer size)
	ros::Publisher pub = nh.advertise<std_msgs::String>("/robot_news_radio", 10);

	// create a rate object which make the loop take 3sec before going to next cycle
	ros::Rate rate(3);

	// while ros node is running
	while (ros::ok()) {
		// create an object
		std_msgs::String msg;

		msg.data = "Hi, this is Radio Mirchi!!";
		pub.publish(msg);
		rate.sleep();

	}

}